plugins/catalog-backend/src/ingestion/processors/AnnotateScmSlugEntityProcessor.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Entity, LocationSpec } from '@backstage/catalog-model';
+import parseGitUrl from 'git-url-parse';
+import { identity, merge, pickBy } from 'lodash';
+import { CatalogProcessor } from './types';
+
+const GITHUB_ACTIONS_ANNOTATION = 'github.com/project-slug';
+
+export class AnnotateScmSlugEntityProcessor implements CatalogProcessor {
+  async preProcessEntity(
+    entity: Entity,
+    location: LocationSpec,
+  ): Promise<Entity> {
+    if (entity.kind !== 'Component' || location.type !== 'url') {
+      return entity;
+    }
+
+    const gitUrl = parseGitUrl(location.target);
+    let githubProjectSlug =
+      entity.metadata.annotations?.[GITHUB_ACTIONS_ANNOTATION];
+
+    if (gitUrl.source === 'github.com' && !githubProjectSlug) {
```

Comment:
This class should take an `ScmIntegrationRegistry` constructor parameter (and have a `fromConfig` that makes the `ScmIntegrations.fromConfig` for you), and judge by the integration type of the URL whether to go ahead or not.

I realize that the annotation doesn't support anything but `github.com` anyway - or, rather, it supports _one implicit host_. It's conceivable that an org uses the annotation for their GHE instance. In any case, I think this processor could do well to just use the integrations since they are there, and not make too many assumptions. If you add the processor you probably do it because you want the annotation.

Reply:
Updates to use ScmIntegrations